src/natcap/invest/forest_carbon_edge_effect.py

```diff
@@ -630,33 +631,31 @@ def _calculate_lulc_carbon_map(
 
     """
     # classify forest pixels from lulc
-    biophysical_table = utils.read_csv_to_dataframe(
-        biophysical_table_path, 'lucode').to_dict(orient='index')
+    biophysical_df = utils.read_csv_to_dataframe(
+        biophysical_table_path, MODEL_SPEC['args']['biophysical_table_path'])
 
     lucode_to_per_cell_carbon = {}
     cell_size = pygeoprocessing.get_raster_info(
         lulc_raster_path)['pixel_size']  # in meters
     cell_area_ha = abs(cell_size[0]) * abs(cell_size[1]) / 10000
 
     # Build a lookup table
-    for lucode in biophysical_table:
+    for lucode, row in biophysical_df.iterrows():
         if compute_forest_edge_effects:
-            is_tropical_forest = (
-                int(biophysical_table[int(lucode)]['is_tropical_forest']))
+            is_tropical_forest = row['is_tropical_forest']
         else:
-            is_tropical_forest = 0
-        if ignore_tropical_type and is_tropical_forest == 1:
+            is_tropical_forest = False
+        if ignore_tropical_type and is_tropical_forest:
             # if tropical forest above ground, lookup table is nodata
-            lucode_to_per_cell_carbon[int(lucode)] = NODATA_VALUE
+            lucode_to_per_cell_carbon[lucode] = NODATA_VALUE
         else:
-            try:
-                lucode_to_per_cell_carbon[int(lucode)] = float(
-                    biophysical_table[lucode][carbon_pool_type]) * cell_area_ha
-            except ValueError:
+            if pandas.isna(row[carbon_pool_type]):
                 raise ValueError(
                     "Could not interpret carbon pool value as a number. "
                     f"lucode: {lucode}, pool_type: {carbon_pool_type}, "
-                    f"value: {biophysical_table[lucode][carbon_pool_type]}")
+                    f"value: {row[carbon_pool_type]}")
```

Comment:
If the value was not a valid number, an error would be raised earlier in `utils.read_csv_to_dataframe`, since it's now validating against the spec for the biophysical table. If we've gotten to this point, `row[carbon_pool_type]` should be guaranteed to be a float or NaN.

Reply:
Aha!  Makes sense, thanks!